cf/mixin/fielddomainlist.py

```diff
@@ -0,0 +1,339 @@
+from ..functions import _DEPRECATION_ERROR_ARG
+
+
+class FieldDomainList:
+    """TODO."""
+
+    def select_by_construct(self, *identities, OR=False, **conditions):
+        """Select elements by their metadata constructs.
+
+        To find the inverse of the selection, use a list comprehension
+        with the !match_by_construct` method of the construct
+        elements. For example, to select all constructs that do *not*
+        have a "latitude" metadata construct:
+
+           >>> gl = cf.{{class}}(
+           ...     f for f in fl if not f.match_by_construct('latitude')
```

Comment:
That looks like a generator expression rather than a list comp. as-is ~~so we should probably add the brackets around it unless the former is in fact required in which case a change of terminology might be suitable?~~ I just tried out the case given above and it errors with `TypeError: cannot create 'generator' instances` so it looks like the brackets are indeed required (tested as working):

```suggestion
           ...     [f for f in fl if not f.match_by_construct('latitude')]
```

Reply:
Interesting - works for me:
```python
In [1]: import cf

In [2]: f = cf.example_field(3)

In [3]: cf.FieldList(x for x in [f] if x.match('precipitation_flux'))

Out[3]: [<CF Field: precipitation_flux(cf_role=timeseries_id(4), ncdim%timeseries(9)) kg m-2 day-1>]

In [4]: 
```

Reply:
Oh, sorry, my bad - it does indeed seem to work :grimacing: . I tested with your example here and also with `[f]` replaced by `cf.FieldList([f, f])`. Looking back at my review notes I was mistakenly calling the method on the `FieldList` rather than the `FieldList` itself.

If it works without the need for list comp. brackets then even better. Nice!